### PR TITLE
#2814 RealizedModeChoiceWriter

### LINKED ISSUE #2814: New York: Create realized mode choices event file
In order to make an analysis easier, instead of reading events.csv.gz (which is pretty large and we don't write it on every iteration), let's create another listener which will write only realized mode choice events.

### PR DESCRIPTION
**Pull request by aaltergot**
Resolves #2814 

This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/2978)

### CODE REVIEW

**Comment by aaltergot** [comment]
test now!

File: src/main/scala/beam/analysis/RealizedModeChoiceWriter.scala [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  }
+
+  override def notifyIterationStarts(event: IterationStartsEvent): Unit = {
+    csvFilePath = beamServices.matsimServices.getControlerIO.getIterationFilename(
+      event.getIteration,
+      "realizedModeChoice.csv.gz"
+    )
+
+    // writeHeader is invoked in constructor
+    csvWriter = new BeamEventsWriterCSV(
```
Comment (rajnikantsh) [inline src/main/scala/beam/analysis/RealizedModeChoiceWriter.scala:64]: possibility  for resource leak , close csv writer for any exceptions

**Review by rajnikantsh — CHANGES_REQUESTED** [review]

File: src/main/scala/beam/analysis/RealizedModeChoiceWriter.scala (hunk 1 above)
Comment (JustinPihony) [reply src/main/scala/beam/analysis/RealizedModeChoiceWriter.scala:64]: This is the last thing done in this method - and closing it on exceptions outside of this won't make a difference. So, the only place needed would be `notifyIterationEnds` as far as I can tell

File: src/main/resources/beam-template.conf [hunk 2]
```diff
@@ -663,6 +663,8 @@ beam.debug {
 
     vmInformation.createGCClassHistogram = "boolean | false"
     writeModeChoiceAlternatives = "boolean | false"
+
+    writeRealizedModeChoiceFile = "boolean | false"
```
Comment (JustinPihony) [inline src/main/resources/beam-template.conf:667]: Should there be anything added to the docs so that this isn't a hidden feature?

**Review by JustinPihony — APPROVED** [review]

File: src/main/resources/beam-template.conf (hunk 2 above)
Comment (aaltergot) [reply src/main/resources/beam-template.conf:667]: I haven't found any mentions of "beam.debug" config in docs. Did I miss it somewhere? (https://github.com/LBNL-UCB-STI/beam/blob/develop/docs/inputs.rst] looks like a correct place. Shall we create a task to document the whole section?

Comment (aaltergot) [reply src/main/resources/beam-template.conf:667]: `beam-template.conf` is also a doc, comments may be added for details. Let's consider `writeRealizedModeChoiceFile ` naming descriptive enough.

**Review by JustinPihony — APPROVED** [review]

**Comment by JustinPihony** [comment]
@rajnikantsh Is this good with you now?

**Comment by rajnikantsh** [comment]
> @rajnikantsh Is this good with you now?

Yep, LGTM

**Review by rajnikantsh — APPROVED** [review]

**Review by REASY — APPROVED** [review]

**Pull request merged**